rosplan_demos_interfaces/rosplan_interface_mapping/include/rosplan_interface_mapping/RPRoadmapServer.h

```diff
@@ -1,170 +1,222 @@
-				point.x = real_x;
-				point.y = real_y;
-				
-				occupancy_grid_utils::Cell cell = occupancy_grid_utils::pointCell(map_meta_data, point);
-				grid_x = cell.x;
-				grid_y = cell.y;
-			}
-		}
-
-		std::string wpID;
-		int grid_x;
-		int grid_y;
-		double real_x;
-		double real_y;
-		std::vector<std::string> neighbours;
-	};
-
-	struct Edge 
-	{
-		Edge(const std::string &s, const std::string &e)
-			: start(s), end(e) {}
-
-		std::string start;
-		std::string end;
-	};
-
-	class RPRoadmapServer
-	{
-
-	private:
-		
-		std::string data_path;
-		std::string static_map_service;
-		bool use_static_map;
-
-		// odometry
-		nav_msgs::OccupancyGrid cost_map;
-		geometry_msgs::PoseStamped base_odom;
-		ros::ServiceClient map_client;
-		tf::TransformListener tf;
-
-		// Scene database
-		mongodb_store::MessageStoreProxy message_store;
-
-		// Knowledge base
-		ros::ServiceClient update_knowledge_client;
-
-		// Roadmap
-		std::map<std::string, Waypoint*> waypoints;
-		std::map<std::string, std::string> db_name_map;
-		std::vector<Edge> edges;
-		bool clearWaypoint(const std::string &id);
-
-		// visualisation
-		ros::Publisher waypoints_pub;
-		ros::Publisher edges_pub;
-		void publishWaypointMarkerArray(ros::NodeHandle nh);
-		void publishEdgeMarkerArray(ros::NodeHandle nh);
-		void clearMarkerArrays(ros::NodeHandle nh);
-		
-		/**
-		 * Check if two waypoints can be connected without colliding with any known scenery. The line should not
-		 * come closer than @ref{min_width} than any known obstacle.
-		 * @param w1 The first waypoint.
-		 * @param w2 The second waypoint.
-		 * @param threshold A value between -1 and 255 above which a cell is considered to be occupied.
-		 * @return True if the waypoints can be connected, false otherwise.
-		 */
-		bool canConnect(const geometry_msgs::Point& w1, const geometry_msgs::Point& w2, int threshold) const;
-
-
-	public:
-
-		/* constructor */
-		RPRoadmapServer(ros::NodeHandle &nh);
-
-		/* callbacks to maps and odom */
-		void odomCallback( const nav_msgs::OdometryConstPtr& msg );
-		void costMapCallback( const nav_msgs::OccupancyGridConstPtr& msg );
-
-		/* service to (re)generate waypoints */
-		bool generateRoadmap(rosplan_knowledge_msgs::CreatePRM::Request &req, rosplan_knowledge_msgs::CreatePRM::Response &res);
-		bool addWaypoint(rosplan_knowledge_msgs::AddWaypoint::Request &req, rosplan_knowledge_msgs::AddWaypoint::Response &res);
-		bool removeWaypoint(rosplan_knowledge_msgs::RemoveWaypoint::Request &req, rosplan_knowledge_msgs::RemoveWaypoint::Response &res);
-
-		void createPRM(nav_msgs::OccupancyGrid map, unsigned int nr_waypoints, double min_distance, double casting_distance, double connecting_distance, int occupancy_threshold, int total_attempts);
-	};
+    struct Waypoint
+    {
+        Waypoint(const std::string &id, unsigned int xCoord, unsigned int yCoord, const nav_msgs::MapMetaData& map_meta_data)
+            : wpID(id), grid_x(xCoord), grid_y(yCoord) {
```

Comment:
I believe code should be moved to the cpps, it's not nice to have it between cpp and .h

Reply:
feedback applied, thanks ;)